plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/metrics/OTelMetricsTelemetry.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.metrics;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.telemetry.OTelTelemetryPlugin;
+
+import java.io.Closeable;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.DoubleCounter;
+import io.opentelemetry.api.metrics.DoubleUpDownCounter;
+import io.opentelemetry.api.metrics.Meter;
+
+/**
+ * OTel implementation for {@link MetricsTelemetry}
+ */
+public class OTelMetricsTelemetry implements MetricsTelemetry {
+    private static final Logger logger = LogManager.getLogger(OTelMetricsTelemetry.class);
+    private final OpenTelemetry openTelemetry;
+    private final Meter otelMeter;
+    private final Closeable metricsProviderClosable;
+
+    /**
+     * Creates OTel based {@link MetricsTelemetry}.
+     * @param openTelemetry OpenTelemetry instance
+     * @param meterProviderCloseable closable to close the meter.
+     */
+    public OTelMetricsTelemetry(OpenTelemetry openTelemetry, Closeable meterProviderCloseable) {
```

Comment:
```suggestion
    public OTelMetricsTelemetry(OpenTelemetrySdk openTelemetry) {
```

Reply:
Actually, OpenTelemetrySdk is a final class and can't be mocked so hard time writing the test cases.

Reply:
(you could even simplify it and use `SdkMeterProvider sdkMeterProvider`)

Reply:
That's also final.

Reply:
I want to avoid the casting inside OTelMetricsTelemetry class. We can pass OpenTelemetry or MeterProvider with the closable as all the MeterProvider implementations might not implement the closable. Your thought here?

Reply:
Oh, sad those are final, but not an issue, we could just do that:

```
public class OTelMetricsTelemetry<T extends MeterProvider & Closeable> implements MetricsTelemetry {
    private static final Logger logger = LogManager.getLogger(OTelMetricsTelemetry.class);
    private final T meterProvider;
    private final Meter otelMeter;

    /**
     * Creates OTel based {@link MetricsTelemetry}.
     * @param openTelemetry OpenTelemetry instance
     * @param meterProviderCloseable closable to close the meter.
     */
    public OTelMetricsTelemetry(T meterProvider, Closeable meterProviderCloseable) {
        this.meterProvider = meterProvider;
        this.otelMeter = openTelemetry.get(OTelTelemetryPlugin.INSTRUMENTATION_SCOPE_NAME);
    }

    @Override
    public void close() throws IOException {
        sdkMeterProvider.close();
    }   
```

(let's make `MetricsTelemetry` implement `Closeable` without overriding `close` methods)


Same works for `OTelTracingTelemetry`:

```
public class OTelTracingTelemetry<T extends OpenTelemetry & Closeable> implements TracingTelemetry {

    private static final Logger logger = LogManager.getLogger(OTelTracingTelemetry.class);
    private final T openTelemetry;
    private final io.opentelemetry.api.trace.Tracer otelTracer;

```

Reply:
Let me try this

Reply:
OTelTracingTelemetry still requires an instance of OpenTelemetry to get the propagators. Should we create the constructor like this?

```
public OTelTracingTelemetry(T tracerProvider, OpenTelemetry openTelemetry, Closeable tracerProviderCloseable) {
        this.otelTracer = tracerProvider.get(OTelTelemetryPlugin.INSTRUMENTATION_SCOPE_NAME);
        this.openTelemetry = openTelemetry;
        this.tracerProviderClosable = tracerProviderCloseable;
    }
```

Reply:
I just posted a snippet above, you get `OpenTelemetry` and `Closeable`:

```
public class OTelTracingTelemetry<T extends OpenTelemetry & Closeable> implements TracingTelemetry {

    private static final Logger logger = LogManager.getLogger(OTelTracingTelemetry.class);
    private final T openTelemetry;
    private final io.opentelemetry.api.trace.Tracer otelTracer;
```

Reply:
Oh, my bad.